### PR TITLE
Chore cleanup turbo functions

### PR DESCRIPTION
**Pull request by EvertBunschoten**
## Proposed Changes
Cleaned up the Riemann boundary condition in CEulerSolver

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [X] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [X] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [X] I used the pre-commit hook to prevent dirty commits and used `pre-commit run --all` to format old commits.
- [X] I have added a test case that demonstrates my contribution, if necessary.
- [x] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

**Comment by pcarruscag** [comment]
@joshkellyjak please review this PR when you have a sec

File: SU2_CFD/src/solvers/CEulerSolver.cpp [hunk 1]
```diff
@@ -5023,109 +5023,76 @@ void CEulerSolver::BC_Riemann(CGeometry *geometry, CSolver **solver_container,
...
-  S_boundary = new su2double[8];
-
-  P_Tensor = new su2double*[nVar];
-  invP_Tensor = new su2double*[nVar];
-  Jacobian_i = new su2double*[nVar];
-  for (iVar = 0; iVar < nVar; iVar++)
-  {
+  const bool viscous              = config->GetViscous(),
+             implicit             = (config->GetKind_TimeIntScheme() == EULER_IMPLICIT),
+             gravity = config->GetGravityForce(),
```
Comment (joshkellyjak) [inline SU2_CFD/src/solvers/CEulerSolver.cpp:5030]: Consistent indentation

File: SU2_CFD/src/solvers/CEulerSolver.cpp [hunk 2]
```diff
@@ -5023,109 +5023,76 @@ void CEulerSolver::BC_Riemann(CGeometry *geometry, CSolver **solver_container,
...
-  su2double *Velocity_b, Velocity2_b, Enthalpy_b, Energy_b, StaticEnergy_b, Density_b, Kappa_b, Chi_b, Pressure_b, Temperature_b;
-  su2double *Velocity_e, Velocity2_e, VelMag_e, Enthalpy_e, Entropy_e, Energy_e = 0.0, StaticEnthalpy_e, StaticEnergy_e, Density_e = 0.0, Pressure_e;
-  su2double *Velocity_i, Velocity2_i, Enthalpy_i, Energy_i, StaticEnergy_i, Density_i, Kappa_i, Chi_i, Pressure_i, SoundSpeed_i;
-  su2double ProjVelocity_i;
-  su2double **P_Tensor, **invP_Tensor, *Lambda_i, **Jacobian_b, **Jacobian_i, **DubDu, *dw, *u_e, *u_i, *u_b;
-  su2double *gridVel, *Residual;
-  su2double *V_boundary, *V_domain, *S_boundary, *S_domain;
 
-  bool implicit             = (config->GetKind_TimeIntScheme() == EULER_IMPLICIT);
   string Marker_Tag         = config->GetMarker_All_TagBound(val_marker);
```
Comment (joshkellyjak) [inline SU2_CFD/src/solvers/CEulerSolver.cpp:5027]:
```suggestion
  const string Marker_Tag         = config->GetMarker_All_TagBound(val_marker);
```

File: SU2_CFD/src/solvers/CEulerSolver.cpp (hunk 1 above)
Comment (EvertBunschoten) [reply SU2_CFD/src/solvers/CEulerSolver.cpp]: Fixed indentation

File: SU2_CFD/src/solvers/CEulerSolver.cpp [hunk 3]
```diff
@@ -5497,7 +5427,6 @@ void CEulerSolver::BC_Riemann(CGeometry *geometry, CSolver **solver_container,
         LinSysRes.SubtractBlock(iPoint, residual);
 
         /*--- Jacobian contribution for implicit integration ---*/
-
         if (implicit)
           Jacobian.SubtractBlock2Diag(iPoint, residual.jacobian_i);
```
Comment (joshkellyjak) [inline SU2_CFD/src/solvers/CEulerSolver.cpp:5432]: Inconsistent line gaps betweeen commments and code in this if statement

File: SU2_CFD/src/solvers/CEulerSolver.cpp [hunk 4]
```diff
@@ -9445,42 +9306,36 @@ void CEulerSolver::TurboAverageProcess(CSolver **solver, CGeometry *geometry, CC
 void CEulerSolver::MixedOut_Average (CConfig *config, su2double val_init_pressure, const su2double *val_Averaged_Flux,
```
Comment (joshkellyjak) [inline SU2_CFD/src/solvers/CEulerSolver.cpp:9306]:
```suggestion
void CEulerSolver::MixedOut_Average(CConfig *config, su2double val_init_pressure, const su2double *val_Averaged_Flux,
```

File: SU2_CFD/src/solvers/CEulerSolver.cpp [hunk 5]
```diff
@@ -9444,42 +9334,36 @@
...
 
   pressure_mix = val_init_pressure;
 
   /*--- Newton-Raphson's method with central difference formula ---*/
-
+  unsigned short iter{0};
   while ( iter <= maxiter ) {
 
-    density_mix = val_Averaged_Flux[0]*val_Averaged_Flux[0]/(val_Averaged_Flux[1] - pressure_mix);
+    const su2double density_mix = val_Averaged_Flux[0]*val_Averaged_Flux[0]/(val_Averaged_Flux[1] - pressure_mix);
```
[the remark this answers is not in the document]
Comment (joshkellyjak) [reply SU2_CFD/src/solvers/CEulerSolver.cpp:9319]: Unsure what the point is in inputting this variable to the function is if it's going to be calculated here? or vice versa

File: SU2_CFD/src/solvers/CEulerSolver.cpp [hunk 6]
```diff
@@ -9445,42 +9306,36 @@ void CEulerSolver::TurboAverageProcess(CSolver **solver, CGeometry *geometry, CC
...
-  su2double relax_factor = config->GetMixedout_Coeff(0);
-  su2double toll = config->GetMixedout_Coeff(1);
+  const auto relax_factor = config->GetMixedout_Coeff(0);
+  const auto toll = config->GetMixedout_Coeff(1);
   unsigned short maxiter = SU2_TYPE::Int(config->GetMixedout_Coeff(2));
-  su2double dhdP, dhdrho, enthalpy_mix, velsq, *vel;
-
-  vel = new su2double[nDim];
 
   pressure_mix = val_init_pressure;
```
Comment (joshkellyjak) [inline SU2_CFD/src/solvers/CEulerSolver.cpp:9313]: Same as density_mix, maybe I'm missing something here?

Comment (EvertBunschoten) [reply SU2_CFD/src/solvers/CEulerSolver.cpp:9313]: density_mix is computed based on the value of pressure_mix, which changes throughout the iterative process. Therefore, a local instance of density_mix can be defined within the scope of the iterative loop.

Comment (joshkellyjak) [reply SU2_CFD/src/solvers/CEulerSolver.cpp:9313]: Yes, I'm being stupid I see now. Unsure why the CodeQL is piping up. Maybe it wants su2double &density_mix as the function argument.

**Review by joshkellyjak — APPROVED** [review]
Looks good to me

**Pull request merged**